crates/symbolicator/src/endpoints/multipart.rs

```diff
@@ -0,0 +1,47 @@
+//! Helper utilities to handle HTTP multipart bodies.
+
+use std::io::SeekFrom;
+
+use axum::extract::multipart::Field;
+use axum::http::StatusCode;
+use futures::prelude::*;
+use tokio::fs::File;
+use tokio::io::AsyncSeekExt;
+use tokio_util::io::StreamReader;
+
+use super::ResponseError;
+
+/// Stream a multipart body to a file.
+///
+/// The file's cursor will be set to the beginning of the file again.
+pub async fn stream_multipart_file(field: Field<'_>, file: &mut File) -> Result<(), ResponseError> {
+    let mut field_reader =
+        StreamReader::new(field.map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err)));
```

Comment:
aha! So there was a `map_err` after all ;-)

Reply:
`futures::...::TryStreamExt`!  i suddenly recalled that should exist